cmd/source_profile.go

```diff
@@ -55,27 +56,45 @@ type SourceProfileConnectionMySQL struct {
 	pwd  string // Same as MYSQLPWD environment variable
 }
 
-func NewSourceProfileConnectionMySQL(params map[string]string) SourceProfileConnectionMySQL {
-	// TODO: Move parsing of environment variables in this function.
+func NewSourceProfileConnectionMySQL(params map[string]string) (SourceProfileConnectionMySQL, error) {
 	mysql := SourceProfileConnectionMySQL{}
-	if host, ok := params["host"]; ok {
-		mysql.host = host
-	}
-	if port, ok := params["port"]; ok {
-		mysql.port = port
-	} else { // Set default port for mysql, which rarely changes.
-		mysql.port = "3306"
+	host, hostOk := params["host"]
+	user, userOk := params["user"]
+	db, dbOk := params["db_name"]
+	port, portOk := params["port"]
+	pwd, pwdOk := params["password"]
+	if !(hostOk || userOk || dbOk || portOk || pwdOk) {
```

Comment:
Maybe add a comment saying on the lines of that we don't users to mix and match params from source-profile and environment variables. We try to get all the required params from source-profile first, and if none are set then we try to read the same from environment variables.